### PR TITLE
VMOD public functions

### PR DESCRIPTION
**Pull request by DavidHenvick**
Jira tasks:
https://bradleycapstone.atlassian.net/browse/LV-519?atlOrigin=eyJpIjoiMmQzMjE0ZDY1ZmMxNGFhOGI3ZTc5OGY0OWY1OWZkNWEiLCJwIjoiaiJ9
https://bradleycapstone.atlassian.net/browse/LV-518?atlOrigin=eyJpIjoiMTZjMzllN2MwNDEzNDVhYThlOTQ3MjBhNzVlNzVhMzciLCJwIjoiaiJ9
https://bradleycapstone.atlassian.net/browse/LV-517?atlOrigin=eyJpIjoiMDlmNDExMmQ5NGI3NDZhYWE2N2UzMWJmYjY2OWViYzkiLCJwIjoiaiJ9

Documentation: https://docs.google.com/document/d/1qyV8T-uRBneLjsCzB8W84C3wD3FJ1SjfEBJIeWANhbo/edit?pli=1&tab=t.z7tsfso1cbvo

Needed parts:
Testing scene: Audio Manager Testing Scene
Important scripts/objects: AudioManager and AudioManagerTester

To test, just start scene, click the buttons : )

### CODE REVIEW

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManager.cs [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
+    //reference to self
+    private static AudioManager _audioManager;
+    //dictionary to hold the currently playing to keep the same instance being layered on itself.
+    private Dictionary<EventReference, EventInstance> _audioPairs;
+
+    //sets all needed variables
+    private void Awake()
+    {
+        if (_audioManager != null && _audioManager != this)
+            Destroy(_audioManager.gameObject);
```
Comment (RyanASwanson) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManager.cs:32]: Part of our code documentation states that we should use always use brackets.

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManager.cs [hunk 2]
```diff
@@ -33,4 +54,67 @@ public override void SetUpMainManager()
...
+        if (reference.IsNull)
+        {
+            Debug.LogWarning("NO REFERENCE SOUND!");
+            return default;
+        }
+
+        EventInstance audioEvent;
+
+        if (_audioPairs.ContainsKey(reference))
+            _audioPairs.TryGetValue(reference, out audioEvent);
```
Comment (RyanASwanson) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManager.cs:91]: Same as the previous comment

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/TestingScript/AudioManagerTester.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+using FMOD.Studio;
+using FMODUnity;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AudioManagerTester : MonoBehaviour
```
Comment (RyanASwanson) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/TestingScript/AudioManagerTester.cs:7]: Seeing as this is a testing class I don't have comments, just don't forget to get rid of it when you are done with it after reviews

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManager.cs [hunk 4]
```diff
@@ -33,4 +54,67 @@ public override void SetUpMainManager()
...
+        audioEvent.start();
+        return audioEvent;
+    }
+
+    /// <summary>
+    /// stops an FMOD sound using the reference it takes
+    /// </summary>
+    /// <param name="audioEvent">the sound instance you want ended</param>
+    /// <param name="fade">toggles whether the sound will fade when done playing</param>
+    public void StopAmbientSound(EventInstance instance, bool fade = false)
```
Comment (RyanASwanson) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManager.cs:107]: Booleans begin with words such as 'is' 'does' 'can' etc.

**Review by RyanASwanson — CHANGES_REQUESTED** [review]
Just a few comments about adhering to our code standards

**Review by RyanASwanson — APPROVED** [review]
Yippee 🎊

**Review by AlecatEnjoyer — APPROVED** [review]
I approve

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+using FMOD.Studio;
```
Comment (ASLH68) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs:1]: File header

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+using FMODUnity;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AudioManagerTester : MonoBehaviour
```
Comment (ASLH68) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs:7]: Class XML Comment

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+{
+    [SerializeField] public EventReference[] OneshotSounds;
+    [SerializeField] public EventReference[] AmbientSounds;
+
+    [SerializeField] AudioManager Manager;
```
Comment (ASLH68) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs:12]: add private specifier and update naming to the proper convention `_audioManager`

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    EventInstance _audioEvent;
+
+    public void PlayOneshotSound(int index)
```
Comment (ASLH68) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs:15]: XML Comment

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        Manager.PlayOneShotSound(OneshotSounds[index]);
+    }
+    public void PlayAmbient(int index)
```
Comment (ASLH68) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs:19]: XML Comment

File: Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        _audioEvent = Manager.PlayAmbientSound(AmbientSounds[index]);
+    }
+    public void StopAmbient()
```
Comment (ASLH68) [inline Assets/General/Scripts/Managers/UniversalManagers/MainUniversalManagers/AudioManagment/StoryBeatAudioManager.cs:23]: XML Comment

**Review by ASLH68 — APPROVED** [review]

**Pull request merged**